examples/modular-drivers/build.rs

```diff
@@ -0,0 +1,58 @@
+use regex::Regex;
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+
+use embuild::build;
+use embuild::espidf::ulp_fsm::SystemIncludes;
+
+fn find_ulp_sources<P: AsRef<Path>>(dir: P) -> io::Result<Vec<PathBuf>> {
+    let mut ulp_sources = Vec::new();
+    for entry in fs::read_dir(dir)? {
+        let entry = entry?;
+        let path = entry.path();
+        if path.is_file() {
+            if let Some(extension) = path.extension() {
+                if extension == "S" {
+                    println!(
+                        "cargo:rerun-if-changed={}",
+                        path.as_os_str().to_str().unwrap()
+                    );
+                    ulp_sources.push(path);
+                }
+            }
+        }
+    }
+
+    Ok(ulp_sources)
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    if Regex::new(r"\w+-esp3?2?s?\d?-espidf")
+        .unwrap()
+        .is_match(&std::env::var("TARGET").unwrap())
+    {
+        embuild::espidf::sysenv::output();
+        embuild::espidf::sysenv::env_path();
+        embuild::espidf::sysenv::idf_path();
+
+        let esp_idf_env =
+            PathBuf::from(std::env::var("DEP_MICRO_RDK_EMBUILD_ESP_IDF_PATH").unwrap());
```

Comment:
note that env can be accessed here because they are propagated by micro-rdk/build.rs which always run before this build script.